docs/guides/buckets.mdx

```diff
@@ -0,0 +1,338 @@
+
+Given the nature of time-series data, ReductStore partitions data into blocks. Each block, stored in a separate file, contains a set of records. There are limits to the block size and the number of records it can hold. When a block reaches its limit, a new block is created to store additional data. This method allows ReductStore to store and access data efficiently.
+You can read more about blocks and records in the [How Does It Work?](../how-does-it-work) document.
+
+Unlike other blob storage or file systems, ReductStore has a flat hierarchy. It doesn't use folders or directories. Instead, all data is stored in entries, which are grouped by buckets.
+
+
+## Bucket Settings
+
+Each bucket has settings that determine how data is stored and accessed. These settings include:
+
+- **Quota Type**: This defines the quota type for the bucket. You can set the quota to either NONE or FIFO, with the default being NONE.
+- **Quota Size**: This represents the quota size in bytes. The default is 0, which indicates no quota.
+- **Max Block Size**: This is the maximum allowable size of a block, in bytes. The default size is 64MB.
+- **Maximal Number of Records**: This is the maximum number of records that can be contained in a block. The default number is 256.
+
+### Quota Type
+
+The quota type determines the method of quota enforcement and can be set to either NONE or FIFO.
+
+NONE implies no quota, allowing data to be stored without any restrictions.
+
+FIFO, short for first-in-first-out, enforces the quota by deleting the oldest block to accommodate new data once the quota is reached. ReductStore ensures that the data does not surpass the quota size, refusing to store new data if there's insufficient room.
+
+### Quota Size
+
+The quota size defines the maximum size of the bucket in bytes. It's ignored if the quota type is set to NONE.
+
+### Max Block Size
+
+The max block size determines the maximum size of a block in bytes. Once a block reaches this limit, a new block is created for storing additional data. A record's size isn't restricted by the block size. If a record exceeds the block size, it will be stored in the current block, and a new block will be created for the next record.
+
+The default value is set at 64MB. Typically, there is no need to alter this unless your records exceed the default size. ReductStore pre-allocates the block size during the creation of a new block. As such, it improves performance when multiple records can be stored in a single block.
+
+### Maximal Number of Records
+
+The maximal number of records refers to the maximum number of records that can be stored in a block. When a block reaches this capacity, a new block is created for additional data. The default value is 256. Generally, you don't need to adjust this unless you have numerous small records and an excess of blocks, which could impact search performance.
+
+To optimize search performance, bear in mind that record searching consists of two steps:
```

Comment:
To optimize search performance with the best settings, keep in mind that searching for records involves a two-step process: